### PR TITLE
Change docs to autoclass

### LINKED ISSUE #535: [BUG] Not all of the `FullNodeClient`'s methods are displayed in the API section
## Expected Behavior

API should display all public methods

## Current Behavior

For example `FullNodeClient.deploy_account` is not in the docs

## Possible Solution

Change `.. automethod` to `.. autoclass`

## Steps to Reproduce
Go to the API section and try to find the `deploy_account` method

### PR DESCRIPTION
**Pull request by drknzz**
Closes #535 

Also added `:member-order: bysource` as with the default, alphabetical order, properties were shuffled into methods.
It is possible to [set it as default in docs](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_default_options).

It looks cleaner with all the `_sync` methods at the bottom.

### CODE REVIEW

File: docs/gateway_client.rst [hunk 1]
```diff
@@ -1,80 +1,8 @@
...
-
-
-The possible statuses are:
-
-- **NOT_RECEIVED**: The transaction has not been received yet (i.e., not written to storage).
-- **RECEIVED**: The transaction was received by the operator.
-- **PENDING**: The transaction passed the validation and is waiting to be sent on-chain.
-- **REJECTED**: The transaction failed validation and thus was skipped.
-- **ACCEPTED_ON_L1**: The transaction was accepted on layer 1.
-- **ACCEPTED_ON_L2**: The transaction was accepted on layer 2.
```
Comment (cptartur) [inline docs/gateway_client.rst:35]: Is this documented somewhere?

**Review by cptartur — COMMENTED** [review]
Except for the documentation of transaction statuses, it looks fine to me. Perhaps we should create an issue to document these better somewhere else if it's not done already?

File: docs/gateway_client.rst (hunk 1 above)
Comment (drknzz) [reply docs/gateway_client.rst:35]: https://starknet.io/docs/hello_starknet/intro.html

Comment (cptartur) [reply docs/gateway_client.rst:35]: Okay, makes sense to me. We shouldn't be duplicating docs that exist for starknet already.

cc @war-in @Solpatium

**Review by war-in — APPROVED** [review]

File: docs/gateway_client.rst (hunk 1 above)
Comment (Solpatium) [reply docs/gateway_client.rst:35]: https://starknet.io/docs/hello_starknet/intro.html#interact-with-the-contract is the better url.

Comment (cptartur) [reply docs/gateway_client.rst:35]: I think @drknzz wasn't suggesting to include this link in our docs anyways.

Comment (Solpatium) [reply docs/gateway_client.rst:35]: Ok, it is fine to remove it

**Review by Solpatium — APPROVED** [review]

**Pull request merged**